### ISSUE TITLE
train model using large amount of memory.

### ISSUE THREAD

**Issue by grafael** [Franck-Dernoncourt/NeuroNER#51]
I just started to train a 40M text file and the amount of RAM memory increased too much... I had to kill the application, was consuming 15Gb and 15Gb of swap... is that normal?

**Comment by fsonntag** [comment]
Yeah, NeuroNER loads the whole dataset in the memory including many many mappings (from IDs to tokens and reverse etc.). Batch loading the dataset from disc would be much more memory efficient though.

**Comment by Franck-Dernoncourt** [comment]
Yes NeuroNER memory footprint could be improved :)
